test/StreamJsonRpc.Tests/MarshalableProxyTests.cs

```diff
@@ -579,6 +579,24 @@ public async Task IMarshalable_MarshaledBackAndForth()
         Assert.True(IsExceptionOrInnerOfType<NotSupportedException>(ex));
     }
 
+    [Fact]
+    public async Task OneObjectMarshalledTwiceHasIndependentLifetimes()
+    {
+        // The method we call twice returns the same object each time,
+        // but JsonRpc doesn't recognize this and assigns a unique token and proxy each time.
+        IMarshalable proxy1 = (await this.client.GetMarshalableAsync().WithCancellation(this.TimeoutToken))!;
+        IMarshalable proxy2 = (await this.client.GetMarshalableAsync().WithCancellation(this.TimeoutToken))!;
+
+        await proxy1.DoSomethingAsync();
+        await proxy2.DoSomethingAsync();
+
+        Assert.NotSame(proxy1, proxy2);
+        proxy1.Dispose();
+
+        await proxy2.DoSomethingAsync();
```

Comment:
Isn't this calling `DoSomethingAsync()` on a disposed `ReturnedMarshalable` (not a disposed proxy) since `proxy1.Dispose()` will cause the disposal of `ReturnedMarshalable`?
I find this line confusing. If my assumption above is correct, I would add a comment before this call clarifying that while `proxy2` is not disposed, the remote object is, so the call may or may not work.

Reply:
The original object's Dispose method doesn't break the object, which is intentional because the point of the test isn't to verify the object, but the RPC connection.
I've added code comments to clarify this.

Reply:
I also added a block to ensure the dispose happened end-to-end before proceeding to test the second proxy.